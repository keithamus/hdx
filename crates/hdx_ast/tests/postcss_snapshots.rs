mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
