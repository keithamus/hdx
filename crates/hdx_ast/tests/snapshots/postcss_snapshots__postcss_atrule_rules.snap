---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "SupportsRule",
      "at_keyword": {
        "kind": "AtKeyword",
        "offset": 0,
        "len": 9
      },
      "condition": {
        "type": "Is",
        "value": {
          "Property": [
            {
              "kind": "LeftParen",
              "offset": 10,
              "len": 1
            },
            {
              "type": "property",
              "name": {
                "kind": "Ident",
                "offset": 11,
                "len": 14
              },
              "colon": {
                "kind": "Colon",
                "offset": 25,
                "len": 1
              },
              "value": {
                "values": [
                  {
                    "kind": "Whitespace",
                    "offset": 26,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 27,
                    "len": 4
                  }
                ]
              },
              "important": null,
              "semicolon": null
            },
            {
              "kind": "RightParen",
              "offset": 31,
              "len": 1
            }
          ]
        }
      },
      "block": {
        "open": {
          "kind": "LeftCurly",
          "offset": 33,
          "len": 1
        },
        "rules": [
          {
            "type": "AtRule",
            "name": {
              "kind": "AtKeyword",
              "offset": 39,
              "len": 17
            },
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 56,
                  "len": 1
                },
                {
                  "kind": "Ident",
                  "offset": 57,
                  "len": 4
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 61,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 62,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 72,
                    "len": 4
                  },
                  "colon": null,
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 78,
                        "len": 1
                      },
                      {
                        "kind": "Whitespace",
                        "offset": 79,
                        "len": 12
                      },
                      {
                        "kind": "Ident",
                        "offset": 91,
                        "len": 5
                      },
                      {
                        "kind": "Colon",
                        "offset": 96,
                        "len": 1
                      },
                      {
                        "kind": "Whitespace",
                        "offset": 97,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 98,
                        "len": 5
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 103,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 113,
                "len": 1
              }
            }
          },
          {
            "type": "stylerule",
            "selectors": [
              {
                "components": [
                  {
                    "type": "tag",
                    "value": {
                      "Unknown": {
                        "kind": "Ident",
                        "offset": 123,
                        "len": 2
                      }
                    }
                  }
                ],
                "comma": null
              }
            ],
            "type": "style-declaration",
            "open": {
              "kind": "LeftCurly",
              "offset": 126,
              "len": 1
            },
            "declarations": [
              {
                "type": "property",
                "name": {
                  "kind": "Ident",
                  "offset": 140,
                  "len": 5
                },
                "colon": {
                  "kind": "Colon",
                  "offset": 145,
                  "len": 1
                },
                "value": {
                  "Named": {
                    "white": {
                      "kind": "Ident",
                      "offset": 147,
                      "len": 5
                    }
                  }
                },
                "important": null,
                "semicolon": null
              }
            ],
            "rules": [],
            "close": {
              "kind": "RightCurly",
              "offset": 161,
              "len": 1
            }
          }
        ],
        "close": {
          "kind": "RightCurly",
          "offset": 167,
          "len": 1
        }
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 172,
        "len": 8
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 180,
            "len": 1
          },
          {
            "type": "Function",
            "name": {
              "kind": "Function",
              "offset": 181,
              "len": 9
            },
            "values": [
              {
                "kind": "Colon",
                "offset": 190,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 191,
                "len": 13
              }
            ],
            "close_paren": {
              "kind": "RightParen",
              "offset": 204,
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 205,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 206,
          "len": 1
        },
        "declarations": [
          {
            "type": "Declaration",
            "name": {
              "kind": "Ident",
              "offset": 210,
              "len": 1
            },
            "colon": {
              "kind": "Colon",
              "offset": 211,
              "len": 1
            },
            "value": {
              "values": [
                {
                  "kind": "Ident",
                  "offset": 212,
                  "len": 13
                }
              ]
            },
            "important": null,
            "semicolon": null
          }
        ],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": []
            },
            "block": {
              "type": "Block",
              "start": 225,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 226,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 232,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 242,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 243,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 244,
                        "len": 6
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 250,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 254,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 256,
          "len": 1
        }
      }
    }
  ]
}
